### PR TITLE
Replacing Thread.sleep with Fluent wait

### PR DESCRIPTION
**Pull request by kshinde2512**
### Describe your changes :

I worked on the replacing Thread.sleep to Fluent wait

#
### Type of change :

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/Webhooks/WebhooksPageTest.java [hunk 1]
```diff
@@ -103,7 +108,7 @@ void checkDuplicateWebhookName() throws InterruptedException {
     Events.click(webDriver, webhooks.allEntities());
     Events.click(webDriver, webhooks.clickToCloseDropdown());
     Events.click(webDriver, common.saveWebhook());
-    Thread.sleep(waitTime);
+    pause(waitTime);
```
Comment (harshach) [inline openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/Webhooks/WebhooksPageTest.java:111]: @kshinde2512 how is this different than what we had before. We are just moving thread.sleep to a method. Why aren't you using the await? @vivekratnavel

Comment (kshinde2512) [reply openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/Webhooks/WebhooksPageTest.java:111]: @harshach This was just a temporary fix. Fluent wait and await was not working as expected so I had to put this with Fluent wait. But the replacement for this is also found and will merge the changes in PR shortly

File: openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/Webhooks/WebhooksPageTest.java [hunk 2]
```diff
@@ -108,7 +99,7 @@ void checkDuplicateWebhookName() throws InterruptedException {
     Events.click(webDriver, webhooks.allEntities());
     Events.click(webDriver, webhooks.clickToCloseDropdown());
     Events.click(webDriver, common.saveWebhook());
-    pause(waitTime);
+    Events.waitForElementToDisplay(webDriver, webhooks.toast(), 10);
     WebElement errorMessage = webDriver.findElement(webhooks.toast());
     Assert.assertTrue(errorMessage.isDisplayed());
```
Comment (vivekratnavel) [inline openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/Webhooks/WebhooksPageTest.java:104]: Do we need these two lines?

Comment (kshinde2512) [reply openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/Webhooks/WebhooksPageTest.java:104]: @vivekratnavel I have removed the assertions for the same

File: openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/events/Events.java [hunk 3]
```diff
@@ -38,4 +38,27 @@ public static void sendEnter(WebDriver driver, By by) {
...
+
+  public static void sleep(int timeOutInSeconds) {
+    try {
+      Thread.sleep(timeOutInSeconds);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
+  }
+
+  public static boolean waitForElementToDisplay(WebDriver driver, By by, int timeOutInSeconds) {
```
Comment (harshach) [inline openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/events/Events.java:50]: @kshinde2512 why can't we use the option described here https://stackoverflow.com/questions/20903231/how-to-wait-until-an-element-is-present-in-selenium
also, this looks interesting. https://selenide.org/index.html cc @vivekratnavel

File: openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/events/Events.java [hunk 4]
```diff
@@ -38,4 +40,25 @@ public static void sendEnter(WebDriver driver, By by) {
...
+            .ignoring(StaleElementReferenceException.class);
+    for (int i = 0; i < 2; i++) {
+      try {
+        fWait.until(ExpectedConditions.presenceOfElementLocated(by));
+        fWait.until(ExpectedConditions.visibilityOf(driver.findElement(by)));
+        fWait.until(ExpectedConditions.elementToBeClickable(driver.findElement(by)));
+      } catch (Exception e) {
+        System.out.println("Element Not found trying again");
+        e.printStackTrace();
+        fWait.withTimeout(Duration.ofSeconds(2));
```
[the remark this answers is not in the document]
Comment (kshinde2512) [reply openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/events/Events.java:59]: @sureshms I will make those changes

**Comment by kshinde2512** [comment]
@harshach @shahsank3t This will fix all failing Selenium tests except one where Ingestion is not created  error is thrown for DatabaseServicePageTest
Reason for this is broken Airflow APIs, once fixed this should pass as well.

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx [hunk 5]
```diff
@@ -157,9 +156,6 @@ const AddIngestion = ({
     (data?.source.sourceConfig.config as ConfigClass)?.generateSampleData ??
       true
   );
-  const [enableDebugLog, setEnableDebugLog] = useState(
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx:160]: We should not remove any of these changes

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/Steps/ConfigureIngestion.tsx [hunk 6]
```diff
@@ -210,17 +208,6 @@ const ConfigureIngestion = ({
             </p>
             {getSeparator('')}
           </Field>
-          <Field>
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/Steps/ConfigureIngestion.tsx:213]: We should not remove any of this

**Review by vivekratnavel — CHANGES_REQUESTED** [review]

**Comment by vivekratnavel** [comment]
Closing it since we moved to Cypress.

**Pull request closed without merging**